sys/include/net/gnrc/tcp/tcb.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_tcp TCP
+ * @ingroup     net_gnrc
+ * @brief       RIOT's tcp implementation for the gnrc stack
+ *
+ * @{
+ *
+ * @file
+ * @brief       Transmission Control Block definition
+ *
+ * @author      Simon Brummer <simon.brummer@haw-hamburg.de>
+ * @}
+ */
+
+#ifndef GNRC_TCP_TCB_H_
+#define GNRC_TCP_TCB_H_
+
+#include <stdint.h>
+#include <ringbuffer.h>
+#include <xtimer.h>
+#include <mutex.h>
+#include <msg.h>
+#include "net/gnrc.h"
+#include "fsm.h"
+#include "config.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct _transmission_control_block {
+    /* Address Information, stored externally */
+    uint8_t address_family;                    /**< Address Family of local_addr and peer_addr */
+    uint8_t *local_addr;                       /**< local IP address */
+    uint8_t *peer_addr;                        /**< peer IP address */
+    uint16_t *local_port;                      /**< local connections port number */
+    uint16_t *peer_port;                       /**< port connections port number */
```

Comment:
these should be changed to simple `uint16_t `, cause a pointer to that is likely (on most platforms) larger than the stored value!